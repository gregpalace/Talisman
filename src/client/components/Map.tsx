import { relative } from 'path';
import React, { Dispatch } from 'react';
import { connect } from 'react-redux';

import MapRow from './MapRow';

interface IMapState {
  readonly map: IMapState;
  readonly player: IPlayerState;
}

interface IPlayerState {
  readonly hp: number;
  readonly mp: number;
  readonly inventorySlots: number;
  readonly position: number[];
  readonly direction: string;
  readonly spritePosition: string;
  readonly walkIndex: number;
}

interface IMapState {
  readonly tiles: number[];
}

interface IMapProps {
  dispatch: Dispatch<any>;
  tiles: any[];
}

const mapStateToProps = (state: IMapState) => {
  return {
    tiles: state.map.tiles,
  };
};
const Map: React.FC<IMapProps> = (props) => {
  return (
    <div
      style={{
        position: 'relative',
        top: '0px',
        left: '0px',
        width: '1000px', // 800
        height: '500px', // 480
        backgroundColor: 'green',
        border: '4px solid white',
        margin: '10px auto',
      }}
    >
      {props.tiles.map((row: number[], index: number) => (
        <MapRow key={index} tiles={row} />
      ))}
    </div>
  );
};

export default connect(mapStateToProps)(Map);
