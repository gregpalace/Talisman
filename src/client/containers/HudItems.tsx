import React, { useState } from 'react';
import { connect } from 'react-redux';

import HealthBar from '../components/hud/healthBar';
import MagicBar from '../components/hud/magicBar';
import Inventory from '../components/hud/Inventory';

import '../../stylesheets/HUD-Styles/hudItems.scss';
import { IState } from '../../typings/PlayerTypes';

const mapStateToProps = (state: IState) => ({
  hp: state.player.hp,
  mp: state.player.mp,
  percentage: state.player.hp,
  inventorySlots: state.player.inventorySlots,
});

const HudItems = ({ hp, mp, inventorySlots, percentage }: any) => {
  const [inventoryOpen, setInventoryOpen] = useState(false);
  const showInventory = () => {
    setInventoryOpen(!inventoryOpen);
  };
  return (
    <div id='hud-items'>
      <HealthBar hp={hp} percentage={percentage} />
      <MagicBar mp={mp} percentage={percentage} />
      {inventoryOpen ? (
        <Inventory hp={hp} mp={mp} inventorySlots={inventorySlots} />
      ) : null}
      <button id='inventory-button' onClick={showInventory}>
        Show Inventory
      </button>
    </div>
  );
};

export default connect(mapStateToProps)(HudItems);
