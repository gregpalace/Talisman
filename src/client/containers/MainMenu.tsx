import React, { useState } from 'react';

import Flame from '../components/Flame';

import '../../stylesheets/mainMenu.scss';

const MainMenu = () => {
  const [isHovering, setHovering] = useState(false);

  const handleMouseHover = () => {
    return setHovering(!isHovering);
  };

  return (
    <div className='main-menu'>
      <header className='heading'>
        <img id='game-logo' src='../../assets/logo/logo.png' alt='Talisman' />
        <h1>Talisman</h1>
      </header>
      <br />
      <main className='menu-options'>
        <ul>
          <li>
            <a
              href='/game'
              onMouseEnter={handleMouseHover}
              onMouseLeave={handleMouseHover}
            >
              {' '}
              Start Game{' '}
              {isHovering && (
                <span>
                  <Flame />
                </span>
              )}
            </a>
          </li>
          <li>
            <a href='/game'>Continue Game</a>
          </li>
        </ul>
      </main>
    </div>
  );
};

export default MainMenu;
