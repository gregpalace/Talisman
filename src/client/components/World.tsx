import React from 'react';

import Map from './Map';
import Player from './Player';

const World = (props: any) => {
  return (
    <div
    style={{
      position: 'relative',
      width: '800px',
      height: '400px',
      margin: '20px auto'
    }}>
      <Map />
      <Player />
    </div>
  );
}

export default World;