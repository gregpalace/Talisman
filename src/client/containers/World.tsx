import React from 'react';
import { connect } from 'react-redux';
// import * as actions from '../actions/actions';

import Map from '../components/Map';
import Player from '../components/Player';
import { tiles } from '../../data/maps/1/map1';
import store from '../../store';

interface IPlayerState {
  readonly hp: number;
  readonly mp: number;
  readonly position: number[];
  readonly inventorySlots: number;
}

const mapStateToProps = (state: IPlayerState) => ({
  position: state.position,
});

const World = (props: any) => {
  store.dispatch({
    type: 'ADD_TILES',
    payload: {
      tiles,
    },
  });
  return (
    <div
      style={{
        position: 'relative',
        width: '1000px',
        height: '500px',
        margin: '20px auto',
      }}
    >
      <Map />
      <Player />
    </div>
  );
};

export default connect(mapStateToProps)(World);
