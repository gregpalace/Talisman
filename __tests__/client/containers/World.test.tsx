import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Player from '../../../src/client/components/Player';
import Map from '../../../src/client/components/Map';
import { World } from '../../../src/client/containers/World';
import HudItems from '../../../src/client/containers/HudItems';

describe('React unit tests', () => {
  let wrapper: any;

  const props = {
    position: [100, 50],
    hp: 100,
    tiles: [0, 1, 3, 4, 3],
    percentage: 100,
    takeDamage: Function,
  };

  beforeEach(() => {
    wrapper = shallow(<World {...props} />);
  });

  describe('<World />', () => {
    it('Should render correctly', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render a wrapper with styling properties', () => {
      expect(wrapper.find('style')).toBeTruthy();
      expect(wrapper.prop('style').width).toEqual('1000px');
      expect(wrapper.prop('style').height).toEqual('500px');
    });

    it('should render <Map /> and <Player /> and <Hud /> components', () => {
      expect(wrapper.find(Map)).toBeDefined();
      expect(wrapper.find(HudItems)).toBeDefined();
      expect(wrapper.find(Player)).toBeDefined();
    });
  });
});
