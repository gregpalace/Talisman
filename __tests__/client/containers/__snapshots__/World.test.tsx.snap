// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React unit tests <World /> Should render correctly 1`] = `
<div
  style={
    Object {
      "height": "500px",
      "margin": "20px auto",
      "position": "relative",
      "width": "1000px",
    }
  }
>
  <Connect(Map) />
  <Connect(Player) />
  <Connect(Component) />
</div>
`;
